pandas/io/date_converters.py

```diff
@@ -1,35 +1,110 @@
 """This module is designed for community supported date conversion functions"""
+import warnings
+
 import numpy as np
 
-from pandas._libs.tslibs import parsing
+from pandas import to_datetime
 
 
 def parse_date_time(date_col, time_col):
-    date_col = _maybe_cast(date_col)
-    time_col = _maybe_cast(time_col)
-    return parsing.try_parse_date_and_time(date_col, time_col)
+    """
+    Parse columns with dates and times into a single datetime column.
+
+    .. deprecated:: 1.1.0
+       Use pd.to_datetime(date_col + " " + time_col) instead to get a Pandas Series.
+       Use pd.to_datetime(date_col + " " + time_col).to_pydatetime() instead to get
+        a Numpy array.
+    """
+    warnings.warn(
+        """
+        Use pd.to_datetime(date_col + " " + time_col) instead to get a Pandas Series.
+        Use pd.to_datetime(date_col + " " + time_col).to_pydatetime() instead to get a Numpy array.
+""",  # noqa: E501
+        FutureWarning,
+        stacklevel=2,
+    )
+    return to_datetime(date_col + " " + time_col).to_pydatetime()
```

Comment:
are these *exactly* the same? `to_datetime` is much more flexible. I would rather not change the actual impl and just deprecate

Reply:
though if we can show they are the same then this is ok

Reply:
What do you mean with just deprecating the impl, that we keep the original impl code and only add the warning at the beginning? I would be fine with that.

Reply:
I reverted back to the original impl. Now we have the original impl and a warning